docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,236 @@
+
+       .. code-block:: java
+
+        // Course.java
+        @OneToMany(mappedBy = "course", fetch = FetchType.LAZY)
+        @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+        @JsonIgnoreProperties("course")
+        private Set<Exercise> exercises = new HashSet<>();
+
+
+.. _ordered:
+
+  * **Ordered Collection**: When you want to order the collection of objects of the relationship, then always use a ``List``. It is important to note here that there is no inherent order in a database table. One could argue that you can use the ``id`` field for the ordering, but there are edge cases where this can lead to problems. Therefore, for a ordered collection, **always** annotate it with ``@OrderColumn``. An order column indicates to hibernate that we want to order our collection based on a specific column of our data table. By default, the column name it expects is *tablenameS\_order*. For ordered collections, we also recommend that you annotate them with ``CascadeType.ALL`` and ``orphanRemoval = true``. E.g.:
+
+       .. code-block:: java
+
+        //Result.java
+        @OneToMany(mappedBy = "result", cascade = CascadeType.ALL, orphanRemoval = true)
+        @OrderColumn
+        @JsonIgnoreProperties(value = "result", allowSetters = true)
+        @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+        @JsonView(QuizView.Before.class)
+        private List<Feedback> feedbacks = new ArrayList<>();
+
+
+       .. note::
+        Hibernate will take care of the ordering for you but you must create the order column in the database. This is not created automatically!
+
+
+    With ordered collections, you have to be very careful with the way you persist the objects in the database. You must first persist the child object without a relation to the parent object. Then, you recreate the association and persist the parent object. Example of how to correctly persist objects in an ordered collection:
+
+       .. code-block:: java
+
+        // ProgrammingAssessmentService
+        List<Feedback> savedFeedbacks = new ArrayList<>();
+        result.getFeedbacks().forEach(feedback -> {
+           // cut association to parent object
+           feedback.setResult(null);
+           // persist the child object without an association to the parent object. IMPORTANT: Use the object returned from the database!
+           feedback = feedbackRepository.save(feedback);
+           // restore the association to the parent object
+           feedback.setResult(result);
+           savedFeedbacks.add(feedback);
+       });
+       
```

Comment:
Unfortunately the warnings are still here. :( (I assume because this line contains whitespaces and isn't empty.)

Reply:
Should be fixed now: e56ec3f